node/src/components/block_synchronizer/tests.rs

```diff
@@ -1951,3 +1994,526 @@ async fn synchronizer_halts_if_block_cannot_be_made_executable() {
         BlockSynchronizerProgress::Syncing(block_hash, _, _) if block_hash == *block.hash()
     );
 }
+
+fn historical_state(block_synchronizer: &BlockSynchronizer) -> &BlockAcquisitionState {
+    block_synchronizer
+        .historical
+        .as_ref()
+        .unwrap()
+        .block_acquisition_state()
+}
+
+#[tokio::test]
+async fn historical_sync_without_a_deploy_works() {
+    let rng = &mut TestRng::new();
+    let mock_reactor = MockReactor::new();
+    let test_env = TestEnv::random(rng);
+    let peers = test_env.peers();
+    let block = test_env.block();
+    let validator_matrix = test_env.gen_validator_matrix();
+    let validators_secret_keys = test_env.validator_keys();
+    let mut block_synchronizer =
+        BlockSynchronizer::new_initialized(rng, validator_matrix, Default::default());
+
+    // Register block for historical sync
+    assert!(block_synchronizer.register_block_by_hash(*block.hash(), true));
+    assert!(block_synchronizer.forward.is_none());
+    block_synchronizer.register_peers(*block.hash(), peers.clone());
+
+    /* Skip steps HaveBlockHeader, HaveWeakFinalitySignature, HaveBlock */
+
+    let historical_builder = block_synchronizer
+        .historical
+        .as_mut()
+        .expect("Historical builder should have been initialized");
+    historical_builder
+        .register_block_header(block.clone().take_header(), None)
+        .expect("header registration works");
+    historical_builder.register_era_validator_weights(&block_synchronizer.validator_matrix);
+    register_multiple_signatures(
+        historical_builder,
+        block,
+        validators_secret_keys
+            .iter()
+            .take(weak_finality_threshold(validators_secret_keys.len())),
+    );
+    assert!(historical_builder.register_block(block, None).is_ok());
+
+    /* Step HaveGlobalState */
+
```

Comment:
Can you change this comment to give details into what we actually want to check?
```suggestion
    // Check that the block synchronizer starts to acquire global state
```

Reply:
Done